src/main.ts

```diff
@@ -138,10 +139,34 @@ function createIdGenerator(): () => Id {
   }
 }
 
-function inferTSConfig(projectPath: string): boolean {
-  // TODO: Move logic from https://github.com/sourcegraph/sourcegraph/pull/30091
-  // into this function.
-  return false;
+/** Returns the list of directory paths under dir which contain a file
+  * with name matchingName.
+*/
+function listDirsRecursive(dir: string, matchingName: string): string[] {
+  let paths: string[] = [];
+  var basenames = fs.readdirSync(dir);
+  for (const basename of basenames) {
+    const path = dir + '/' + basename;
+    const stat = fs.statSync(path);
+    if (stat.isDirectory()) {
+      paths = paths.concat(listDirsRecursive(path, matchingName));
+    } else if (stat.isFile() && basename == matchingName) {
+      paths.push(dir);
+    }
+  }
+  return paths;
+}
+
+function createTSConfigAndEmitLSIF(dir: string): void {
+  // TypeScript packages on NPM are commonly distributed without a
+	// tsconfig.json file. Instead of skipping indexing altogether,
+	// synthesize some permissive tsconfig.json files.

```

Comment:
Inconsistent spacing.

Reply:
Should be fixable with Prettier, which I recommend running on file save